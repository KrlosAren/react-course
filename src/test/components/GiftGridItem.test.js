import React from 'react'
import { shallow } from "enzyme"
import GifGridItem from '../../components/GifGridItem'


const title = 'un titulo'
const url = 'https://localhost/algo.jpg'

const wrapper = shallow(<GifGridItem title={title} url={url} />)

describe('test in <GifGridItem />', () => {


  test('debe mostrar el component correctamente ', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should be have a paragraph with the tittle', () => {
    const paragraph = wrapper.find('p')
    expect(paragraph.text().trim()).toBe(title)
  })

  test('should be have a image file equal to url and alt of props', () => {
    const img = wrapper.find('img')
    expect(img.props().alt).toBe(title)
    expect(img.props().src).toBe(url)
  })

  test('should be have a  animate__fadeIn', () => {
    const div = wrapper.find('div')
    console.log(div.props().className.includes('animate__fadeIn'));
  })



})
