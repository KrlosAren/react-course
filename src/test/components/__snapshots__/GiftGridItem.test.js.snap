// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GifGridItem /> debe mostrar el component correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
